### PR TITLE
Level up perfomance of LiDAR storage & processing

### PR DESCRIPTION
**Pull request by dorkamotorka**
This PR improves LiDAR data storage in two ways:
- std::vector function reserve() is used, which allocates enough memory in the Free-Store so there is no need for resizing and copying all of the data of the std::vector to the new memory location when this happens
- Lookup Tables for Sine and Cosine of theta(LiDAR angle) are used, because std::sin() and std::cos() are quite costly operation that where repeatedly used in our callbacks.  

I did profile the code myself on my PC and experienced 30-50% less time consumed on these for loops. To give you a sense in seconds - Firstly the for loop took ~140 microseconds, afterwards it took ~60 microseconds.

### CODE REVIEW

**Comment by JanezCim** [comment]
do you need only code review or do i need to test on Gazebo and/or real robot?

**Comment by dorkamotorka** [comment]
Please test both.

File: src/obstacle_points.cpp [hunk 1]
```diff
@@ -108,8 +108,8 @@ void ObstaclePoints::scan_callback(const sensor_msgs::LaserScan::ConstPtr &msg)
 {
     float theta = msg->angle_min;
     float increment = msg->angle_increment;
-    float range_min = msg->range_min;
```
Comment (JanezCim) [inline src/obstacle_points.cpp:111]: why is range_min now taken from ROS params and not from lidar message anymore?

**Review by JanezCim — COMMENTED** [review]
Compiles ok on Melodic, works in Gazebo, need to still test on a magni

**Comment by dorkamotorka** [comment]
I created a Lidar class which can be useful in future if we would modify this node to be able to accept point from multiple lidar sources. Please re-test.

Object is created because we need to access lidar meta data from the callback and a method get_points and not only from the callback.

**Review by JanezCim — APPROVED** [review]
I tested on the robot and semms to work fine along with obstacle avoidance.

I also checked cpu and ram usage with on kinetic-devel branch
kinetic_move_basic

and this branch
performance_move_basic

Seems to consume a bit less CPU but about same amount of RAM

**Pull request merged**